opengate/engines.py

```diff
@@ -315,6 +454,10 @@ def __init__(self, simulation_engine):
         self.g4_em_parameters = None
         self.g4_parallel_world_physics = []
         self.g4_optical_material_tables = {}
+        self.g4_physical_volumes = []
```

Comment:
Please do not store extra references to the G4PhysicalVolumes here. They are accessible via every volume object as volume.g4_physical_volumes.